src/views/Graph.vue

```diff
@@ -558,25 +956,66 @@ export default {
         `)
       }
 
-      if (this.groupCycle) {
-        // Loop over the subgraphs
-        Object.keys(cycles).forEach((key, i) => {
-          // Loop over the nodes that are included in the subraph
-          const nodeFormattedArray = cycles[key].map(a => `"${a.id}"`)
-          ret.push(`
-          subgraph cluster_margin_${i}
-          {
-            margin=100.0
-            label="margin"
-            subgraph cluster_${i} {${nodeFormattedArray};\n
-              label = "${key}";\n
-              fontsize = "70px"
-              style=dashed
-              margin=60.0
-            }
-          }`)
+      const graphSections = {}
+      Object.keys(cycles).forEach((cycle, iCycle) => {
+        const indexSearch = Object.values(this.cylcTree.$index).filter((node) => {
```

Comment:
This is searching over the full `cylcTree.$index`, this is the complete data store which also contains nodes for other workflows.

We only need to iterate over the cycles in this workflow, i.e. `Object.keys(cycles).forEach((key, i) => {`.